middle_end/flambda2.0/to_cmm/un_cps_helper.ml

```diff
@@ -410,34 +403,48 @@ let lambda_ba_kind k =
   | Complex32 -> Lambda.Pbigarray_complex32
   | Complex64 -> Lambda.Pbigarray_complex64
 
-let bigarray_load ?(dbg=Debuginfo.none) (is_safe : Flambda_primitive.is_safe)
-      _dims kind layout = function
-  | ba :: args ->
-      assert (List.length args = _dims); (* TODO: correct ? *)
-      let kind = lambda_ba_kind kind in
-      let layout = lambda_ba_layout layout in
-      let is_unsafe =
-        match is_safe with
-        | Safe -> false
-        | Unsafe -> true
-      in
-      bigarray_get is_unsafe kind layout ba args dbg
-  | _ -> assert false
-
-let bigarray_store ?(dbg=Debuginfo.none) (is_safe : Flambda_primitive.is_safe)
-      _dims kind layout = function
-  | ba :: args ->
-      let indexes, value = Misc.split_last args in
-      let kind = lambda_ba_kind kind in
-      let layout = lambda_ba_layout layout in
-      let is_unsafe =
-        match is_safe with
-        | Safe -> false
-        | Unsafe -> true
-      in
-      return_unit dbg (
-        bigarray_set is_unsafe kind layout ba indexes value dbg)
-  | _ -> assert false
+let bigarray_load ?(dbg=Debuginfo.none) _dims kind _layout ba offset =
+  let elt_kind = lambda_ba_kind kind in
+  let elt_size = bigarray_elt_size elt_kind in
+  let elt_chunk = bigarray_word_kind elt_kind in
+  let ba_data_f = field_address ba 1 dbg in
+  let ba_data_p = load ~dbg Cmm.Word_int Asttypes.Mutable ba_data_f in
```

Comment:
You shouldn't need to qualify with `Asttypes` here (neither `Cmm` or `Lambda` below).